### PR TITLE
use iterator instead of vec to iterate accounts

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

avoid creating a vector just to iterate

#### Summary of Changes

create an iterator over contents of an append vec
a follow on pr will replace 2 more sites with this.
This particular caller is already in code that is very expensive at high account counts.

Fixes #

### CODE REVIEW

File: runtime/src/append_vec.rs [hunk 1]
```diff
@@ -136,6 +136,33 @@ impl<'a> StoredAccountMeta<'a> {
     }
 }
 
+pub struct AccountsInAppendVec<'a> {
```
Comment (brooksprumo) [inline runtime/src/append_vec.rs:139]: How about calling this something like `AppendVecAccountsIter`?

**Review by brooksprumo — DISMISSED** [review]
Lgtm

**Review by brennanwatt — APPROVED** [review]

File: runtime/src/append_vec.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/append_vec.rs]: done

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -5413,8 +5415,7 @@ impl AccountsDb {
         let mut progress = Vec::with_capacity(len);
         let mut current = Vec::with_capacity(len);
         for storage in storages {
-            let accounts = storage.accounts.accounts(0);
```
Comment (jeffwashington) [inline runtime/src/accounts_db.rs:5416]: this allocates a Vec<>

**Review by apfitzge — APPROVED** [review]
looks good, nice change!

**Pull request merged**